### PR TITLE
Add and Enable Loop Restoration Filters

### PR DESCRIPTION
**Pull request by xiphmont**
Work in progress, currently here for the automated testing.

### CODE REVIEW

File: src/context.rs [hunk 1]
```diff
@@ -3029,7 +3029,7 @@ impl ContextWriter {
                   symbol_with_update!(self, w, 0, &mut self.fc.lrf_sgrproj_cdf);
                 }
```
Comment (ycho) [inline src/context.rs]: Indentation, need move left one col, from here to line 3035. :)

File: src/context.rs [hunk 2]
```diff
@@ -3041,7 +3041,8 @@ impl ContextWriter {
                   symbol_with_update!(self, w, 1, &mut self.fc.lrf_sgrproj_cdf);
                 }
```
Comment (ycho) [inline src/context.rs]: Need fix indents from here downto 3048 as well?

File: src/lrf.rs [hunk 3]
```diff
@@ -55,13 +63,257 @@ pub const SGRPROJ_PARAMS_EPS: [[u8; 2]; 1 << SGRPROJ_PARAMS_BITS] = [
...
+        acc += hfilter[i as usize] * src_plane.p(clipped_cfg.width - 1, ly) as i32;
+      }
+        
+      acc = acc + (1 << round_h >> 1) >> round_h;
+      work[(yi-y+3) as usize] = clamp(acc, -offset, limit-offset);
+    }
+
+    for (wi, dst) in (start_wi..start_wi+end_i).zip(out_data[xi..].iter_mut().step_by(stride).take(end_i)) {
+      let mut acc = 0;
+      for (i,src) in (0..7).zip(work[wi..wi+7].iter_mut()) {
```
Comment (lu-zero) [inline src/lrf.rs]: Maybe
``` rust
for (vf, &src) in vfilter[0..7].iter().zip(work[wi..wi+7].iter()) {
  acc += vf * src;
}
```

Comment (xiphmont) [reply src/lrf.rs]: Oh, ha, yeah.  Will simplify in an upcoming patch :-)

**Pull request merged**